### ISSUE TITLE
'Dummy' audio device actually uses ALSA backend

### ISSUE THREAD

**Issue by Wallacoloo** [LMMS/lmms#2349]
**Steps to reproduce**:
1. Open Edit->Settings and configure a working ALSA backend
2. Restart LMMS (you should have working sound)
3. Open Edit->Settings and switch to the 'Dummy' audio device
4. Restart LMMS.

After the above steps, you should actually have working sound output. The 'Dummy' device is supposed to have no sound output, but the ALSA backend is actually being opened instead.

This can be confirmed by repeating the steps but setting the ALSA device to 'null' in step 1. Now the dummy output will actually be silent. It can also be confirmed by putting a breakpoint in AudioAlsa::run - it will be hit even when using the dummy device.

**Comment by Wallacoloo** [comment]
The code responsible for this appears to be [Mixer::tryAudioDevices](https://github.com/LMMS/lmms/blob/master/src/core/Mixer.cpp#L739):
```c++
AudioDevice * Mixer::tryAudioDevices()
{
	bool success_ful = false;
	AudioDevice * dev = NULL;
	QString dev_name = ConfigManager::inst()->value( "mixer", "audiodev" );

	if( dev_name == AudioDummy::name() )
	{
		dev_name = "";
	}

#ifdef LMMS_HAVE_ALSA
	if( dev_name == AudioAlsa::name() || dev_name == "" )
	{
		dev = new AudioAlsa( success_ful, this );
		if( success_ful )
		{
			m_audioDevName = AudioAlsa::name();
			return dev;
		}
		delete dev;
	}
#endif
...
}
```
which is reached via the following code path:
```
#0  0x0000000000520dea in Mixer::tryAudioDevices() ()
#1  0x000000000051f3d2 in Mixer::initDevices() ()
#2  0x00000000004ff15c in Engine::init(bool) ()
#3  0x00000000005869e4 in GuiApplication::GuiApplication() ()
#4  0x00000000004d67b6 in main ()
```

It's coded explicitly to where the "dummy (no audio output)" device setting is interpreted to mean "automatically choose a working device". Obviously, that's incorrect. I'm guessing this is a result of LMMS trying to choose a sane output default on first run?

**Comment by tresf** [comment]
This may seem like a bug at first glance, but it really is an enhancement.  I think what you're looking for is **Ability to store Dummy backend** which currently is synonymous with no-backend.

The question is.... when our users fallback to "Dummy" because a valid backend isn't found, do we accept the default behavior and allow the software to try again next time the software is opened, or do we force Dummy.

The best approach would probably be a dedicated setting that signifies the user actually selected "Dummy" manually, but I don't believe the current code checks for a "dirty" settings dialog before saving, so there's currently no difference between leaving it dummy and explicitly selecting dummy.

**Comment by Wallacoloo** [comment]
@tresf I've removed the 'core' label.

I'm fine with whatever behavior you/others think is best for "Dummy". The issue I have is with "Dummy" doing something different than labeled. In the settings dialog, the dummy device is explicitly labeled "dummy (no sound output)". To then have the dummy device *produce sound output* is a direct contradiction of its description.

![lmms-dummy-audio](https://cloud.githubusercontent.com/assets/1210751/9974783/2cf1dbce-5e56-11e5-826e-b62bf07109a4.png)

Should we just remove the "(no sound output)" text from the description then? Maybe change it to "dummy/autodetect"?

**Comment by tresf** [comment]
> dummy device is explicitly labeled "dummy (no sound output)". To then have the dummy device produce sound output is a direct contradiction of its description.

Oh.  Then I agree that is a bug.  So are we just not updating the GUI after the correct device is auto-selected?

**Comment by M374LX** [comment]
Fixed by #2376?

**Comment by tresf** [comment]
> Fixed by #2376?

:+1:
